### ISSUE TITLE
MD025: The check that a page has only one level 1 heading doesn't work if one of the headings is in YAML frontmatter

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#450]
### rumdl version

0.1.25

### Installation method

mise

### Bug description

`rumdl` has check MD025 (https://github.com/rvben/rumdl/blob/ee659eacb863b89d32cb8c64833e9af3c56daa61/docs/md025.md) that makes sure that a Markdown file has only one level 1 heading. It also seems to have a customisable `level` setting, though it's not clear why because Markdown files will very often have more than one level 2 heading etc.

The check seems to work fine if multiple level 1 headings are present in the main Markdown document. For example:

```markdown
# Heading in document 1

# Heading in document 2

Some introductory text.
```

In this case, `rumdl` will flag line 3 as a violation. However, if a level 1 header is declared once in the frontmatter and once in the main document, MD025 will not flag it as a violation even though this means there are two level 1 headings.

### Steps to reproduce

1. Create a Markdown file with this content:
   ```markdown
   ---
   title: Heading in frontmatter
   ---

   # Heading in document

   Some introductory text.
   ```
1. Run `rumdl` with no special configuration to make use of default settings.

### Expected behavior

`rumdl` should flag the level 1 heading in the main document as a second level 1 heading because of the level 1 heading already defined in the YAML frontmatter.

### Actual behavior

`rumdl` finds no violations.

### Configuration

None. Bug manifests when using default configuration.

### Operating system

macOS

**Comment by eread** [comment]
@rvben Thanks for fixing this. Seems to work well in version `0.1.26`.

**Comment by rvben** [comment]
Thanks @eread, this is fixed in v0.1.26. The frontmatter `title` field now counts as the first H1.
